wallet/src/de/schildbach/wallet/ui/main/MainActivity.kt

```diff
@@ -401,20 +344,9 @@ class MainActivity : AbstractBindServiceActivity(), ActivityCompat.OnRequestPerm
         upgradeWalletCoinJoin(true)
     }
 
-    private fun handleInvite(invite: InvitationLinkData, isLocked: Boolean, isSynced: Boolean) {
-        if (isLocked || !isSynced) {
-            log.info("handling invite: wait for sync ($isSynced) or unlock ($isLocked)")
-            viewModel.pendingInvite = invite
-        } else {
-            log.info("handling invite: ${invite.link}")
-            val inviteHandler = InviteHandler(this, viewModel.analytics)
-            inviteHandlerViewModel.inviteData.observe(this) {
-                inviteHandler.handle(it, false) {
-                    val acceptInviteIntent = AcceptInviteActivity.createIntent(this, invite, false)
-                    startActivity(acceptInviteIntent)
-                }
-            }
-            inviteHandlerViewModel.handleInvite(invite)
+    private fun handleInvite(invite: InvitationLinkData) {
+        lifecycleScope.launch {
+            inviteHandlerViewModel.setInvitationLink(invite, false)
         }
```

Comment:
invite handling is simplified.  the WalletTransactionsFragment is what displays UI for it.